main.go

```diff
@@ -333,7 +349,11 @@ func serve(logger log.Logger, addr string, writers []writer, readers []reader) e
 		level.Debug(logger).Log("msg", "/read", "duration", duration) //nolint:errcheck
 	})
 
-	return http.ListenAndServe(addr, nil)
+	if err := srv.ListenAndServe(); err != http.ErrServerClosed {
+		level.Error(logger).Log("msg", "Failed to listen", "addr", addr, "err", err) //nolint:errcheck
```

Comment:
@wdoogz do we need to exit here if the web server fails to start listening?

Reply:
Looks like the exit used to be here before your change ... https://github.com/KohlsTechnology/prometheus_bigquery_remote_storage_adapter/pull/59/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261L144

Reply:
I had to change the error handling from inside of the main function to inside of the serve function, and create a http server to be able to gracefully shutdown the server with this line.

https://github.com/KohlsTechnology/prometheus_bigquery_remote_storage_adapter/pull/59/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R243

I should add an exit at this line though, good catch.  I should add one at line 244 too.